MekHQ/src/mekhq/campaign/parts/TankLocation.java

```diff
@@ -104,8 +103,8 @@ protected void computeCost () {
     @Override
     public boolean isSamePartType(Part part) {
         return part instanceof TankLocation
-                && getLoc() == ((TankLocation)part).getLoc()
-                && getUnitTonnage() == ((TankLocation)part).getUnitTonnage();
+                && getLoc() == ((TankLocation) part).getLoc()
+                && getUnitTonnage() == part.getUnitTonnage();
```

Comment:
Double check that this doesn't have an implementation of getUnitTonnage()

Reply:
Same as above